protos/guard/guard.proto

```diff
@@ -0,0 +1,188 @@
+    LOST = 4; //some shards were lost and the whole file cannot be recovered now
+    DONE = 5; //all contracts were done
+  }
+  MetaState state = 2;
+  repeated Contract contracts = 3;
+  bytes renter_signature = 4; //renter's signature for FileStoreMeta and all contractMeta
+  google.protobuf.Timestamp guard_receive_time = 5; //guard received time, which used to calculate the timeout for cancel
+  repeated Log change_log = 6;
+  google.protobuf.Timestamp curent_time = 7; //the time for generate the status
+  bytes guard_signature = 8;
+}
+
+message ChallengeQuestion {
+  bytes shard_hash = 1; //shard hash
+  bytes host_address = 2; //host's address
+  int32 chunk_index = 3; //the chunk's index for calculation
+  int32 random_nounce = 4; //the challenge questions's random nounce
+  bytes expect_answer = 5; //the challenge expected answer
+}
+
+message ProofOfReplicateChallenge {
+  ChallengeQuestion question = 1; //of course expect_answer should be clear before send to the host`
+  google.protobuf.Timestamp challenge_time = 2; //challenge time from guard
+  bytes guard_address = 3; //guar's address
+  bytes guard_signature = 4; //guard signed for the challenge
+  bytes host_answer = 5; //challenge answer
+  google.protobuf.Timestamp host_sign_time = 6; //response time from host
+  bytes host_signature = 7;
+}
+
+message FileChallengeQuestions {
+  bytes file_hash = 1;
+  repeated ChallengeQuestions shard_questions = 2;
+}
+
+message ChallengeQuestions {
+  bytes file_hash = 1; //file's hash
+  bytes shard_hash = 2;
+  bytes preparer_address = 3; //challenge questions preparer's address, it is renter address at phase 1
+  int32 request_count = 4;
+  repeated ChallengeQuestion questions = 5;
+  google.protobuf.Timestamp prepare_time = 6;
+  bytes renter_signature = 7;
+}
+
+message CheckFileStoreMetaRequest {
+  bytes file_hash = 1;
+  bytes renter_address = 2;
+  bytes requester_address = 3;
+  bytes signature = 4; //requester signed
+}
+
+enum ResponseCode {
+  SUCCESS = 0;
+  SIGNATURE_FORMAT_ERROR = 1;
+  COMPUTE_ADDRESS_ERROR = 2;
+  OTHER_ERROR = 20;
+}
+
+message Result {
+  ResponseCode code = 1;
+  string message = 2;
+  google.protobuf.Timestamp response_time = 3;
+}
+
+message HealthRequest {
+  bytes request_address = 1;
+  google.protobuf.Timestamp curent_time = 2;
+  bytes signature = 3;
+}
+
+message HealthStatus {
+  bytes peer_id = 1; // peer ID
+  bytes address = 2; // pub key/address
+  enum HealthStatus {
+    WRONG = 0;
+    RUNNING = 1;
+    BOOTSTRAP = 2; //reserve
+    PARTIAL_STOP = 3; //reserve field, some function cannot work
+  }
+  HealthStatus status = 3; //current status , running, starting, partial stop
+  google.protobuf.Timestamp curent_time = 4; //current time of guard
+  google.protobuf.Timestamp start_time = 5;
+  bytes git_hash = 6; //the git hash of the build related the git repository's branch
+  bytes version = 7; //version we defined for current node
+  bytes signature = 8; //signature
+}
```

Comment:
per our sync, it health going to be in a separate PR?  -- I saw its message here, however, rpc service is commented up.

Reply:
Yes, it is better to have another PR as they link to different tasks. But currently it is a little complexity for me to separate them as guard.proto should support serverstatus, that's why I combine them now, otherwise we need do a merge.

Reply:
exactly